Update Vector2's angle method descriptions

mechPenSketch opened the pull request:
Following https://github.com/godotengine/godot-docs/pull/4678, this request updates the affected method descriptions to include illustrated graphs.

This is in continuation of https://github.com/godotengine/godot-docs/issues/4676#issuecomment-784288436.
<!--
Pull requests should always be made for the `master` branch first, as that's
where development happens and the source of all future stable release branches.

Relevant fixes are cherry-picked for stable branches as needed.

Do not create a pull request for stable branches unless the change is already
available in the `master` branch and it cannot be easily cherry-picked.
Alternatively, if the change is only relevant for that branch (e.g. rendering
fixes for the 3.2 branch).
-->

Calinou committed:
Highlight context menu items at the top of the 2D/3D viewports

This makes it easier to notice that some menu items only appear when
specific nodes are selected.

This change applies to both 2D and 3D editors, including both plugin-based
menus and the hardcoded 2D layout/animation contextual menus.

RandomShaper committed:
Rationalize property reversion

akien-mga commented on doc/classes/Vector2.xml:
```diff
@@ -66,6 +66,8 @@
 			<description>
 				Returns this vector's angle with respect to the positive X axis, or [code](1, 0)[/code] vector, in radians.
 				For example, [code]Vector2.RIGHT.angle()[/code] will return zero, [code]Vector2.DOWN.angle()[/code] will return [code]PI / 2[/code] (a quarter turn, or 90 degrees), and [code]Vector2(1, -1).angle()[/code] will return [code]-PI / 4[/code] (a negative eighth turn, or -45 degrees).
+				The following graph illustrates what angle is returned:
+				[img]https://raw.githubusercontent.com/godotengine/godot-docs/master/img/vector2_angle.png[/img]
```
I don't remember if `[img]` actually works in the editor and/or the online docs, but wouldn't it be too much to embed the same graph three times in the docs? An inline link to the graph or the documentation section which includes it might be better?

CC @Calinou @aaronfranke

aaronfranke replied:
@akien-mga This PR adds links to 3 different graphs. I think they're helpful and worth adding, assuming `[img]` works.

akien-mga replied:
Indeed, the CI checks fail too:
```
ERROR: Unresolved type 'img', file: Vector2
ERROR: Unresolved type '/img', file: Vector2
```

mechPenSketch replied:
In hindsight, I should've paid attention to the [class reference writing guidelines](https://docs.godotengine.org/en/latest/community/contributing/class_reference_writing_guidelines.html) whether `[img]` is compatible, but on the other hand, the guidelines make no mention of `[url]` either.

31 committed:
Fix 'script_class' null access when reloading a deleted C# script

Calinou committed:
Make the focus outline translucent for editor viewports

This makes the focus outline less distracting on the
2D and 3D editor viewports.

YeldhamDev committed:
Add `auto_translate` toggle for automatic translation

rsubtil committed:
Implemented initial DAP support

Implemented "output" event

Refactored "seq" field generation

Prevent debugging when editor and client are in different projects

Removed unneeded references to peer on the parser

Refactored way to detect project path

Implemented "setBreakpoints" request

Fix double events when terminating from client

Refactored "stopped" event

Implemented "stopped" with breakpoint event

Implemented "stackTrace", "scopes" and "variables" request

Report incoming number of stack dump variables

Implemented proper reporting of scopes and variables from stack frames

Prevent editor from grabbing focus when a DAP session is active

Implemented "next" and "stepIn" requests

Implemented "Source" checksum computing

Switched expected errors from macros to silent guards

Refactored message_id

Respect client settings regarding lines/columns behavior

Refactored nested DAP fields

Implement reporting of "Members" and "Globals" scopes as well

Fix error messages not being shown, and improved wrong path message

Faless committed:
[Net] Implement lower level ENet wrappers.

aaronfranke committed:
Use real_t in editor plugins

Also use const more often and delete dead code in CanvasItemEditor

Paulb23 committed:
Move auto brace completion to CodeEdit

Paulb23 committed:
Move brace matching into CodeEdit

Paulb23 committed:
Move line length guidelines into CodeEdit

Paulb23 committed:
Move symbol lookup into CodeEdit

Paulb23 committed:
Clean up and complete CodeEdit inspector and docs

fire committed:
Bake RESET animation.

Co-authored-by: MMMaellon <mmmaellon@gmail.com>
Co-authored-by: Eron <rufsketch1@gmail.com>

Faless committed:
[Net] Add generate_unique_id to MultiplayerPeer.

Used by ENetMultiplayerPeer and WebSocketServer to generate network IDs,
and exposed to the user for p2p networks (e.g. WebRTCMultiplayerPeer)
and custom MultiplayerPeer implementations.

Faless committed:
[Net] Refactor ENetMultiplayerPeer to use ENet wrappers.

Calinou committed:
Improve the 2D editor ruler display

- Use the ° symbol instead of "deg" to reduce clutter.
- Round the displayed lengths to only one decimal instead of two
  to further reduce clutter.
- Don't make the `px` suffix localizable, as it isn't localizable
  anywhere else in the editor.

Chaosus committed:
Added more input/output built-ins to visual shaders

SaracenOne committed:
Removing bounding box calculations from 3D scene drag and drop and collide against physics rather than visual geometry.

Razoric480 committed:
Fix LSP reporting wrong types

reduz committed:
Implement Binary Shader Compilation

* Added an extra stage before compiling shader, which is generating a binary blob.
* On Vulkan, this allows caching the SPIRV reflection information, which is expensive to parse.
* On other (future) RenderingDevices, it allows caching converted binary data, such as DXIL or MSL.

This PR makes the shader cache include the reflection information, hence editor startup times are significantly improved.
I tested this well and it appears to work, and I added a lot of consistency checks, but because it includes writing and reading binary information, rare bugs may pop up, so be aware.
There was not much of a choice for storing the reflection information, given shaders can be a lot, take a lot of space and take time to parse.

raulsntos committed:
Ignore paths with invalid chars in PathWhich

akien-mga committed:
Merge pull request #50728 from Chaosus/shader_varying_enchancements

Allow using vertex varying in custom functions under any circumstances

akien-mga committed:
Merge pull request #50729 from Chaosus/shader_varying_enchancements2

Allow using vertex-stage varying in both `fragment` and `light` functions

Chaosus committed:
Fix compilation of `shader_language.cpp`

akien-mga committed:
Merge pull request #48620 from Calinou/editor-3d-hide-selection-box-no-gizmos

Hide the 3D editor selection box when View Gizmos is disabled

akien-mga committed:
Merge pull request #50881 from Chaosus/shader_fix_build

Fix compilation of `shader_language.cpp`

Chaosus committed:
Prevents shader crashing if varying assigned incorrectly

SirQuartz committed:
Fix expression in `cluster_builder_rd.h`

This expression should now work as intended.

akien-mga committed:
Merge pull request #50885 from SirQuartz/patch-33

Fix `if` statement in `cluster_builder_rd.h`

akien-mga committed:
Merge pull request #50884 from Chaosus/fix_shader_crash

Prevents shader crashing if varying assigned incorrectly by using compound assignment operators (*=, += etc.)

vitika9 committed:
Fixed coding style

akien-mga committed:
i18n: Sync translations with Weblate

(cherry picked from commit c406c8512f059eab29a3fc135218b7b60a5315d1)

KoBeWi committed:
Improve ConfigFile example

Chaosus committed:
Fix editor crash if passing index as variable to function parameter

akien-mga committed:
Merge pull request #50847 from reduz/implement-binary-shader-compilation

Implement Binary Shader Compilation

akien-mga committed:
Merge pull request #50895 from Chaosus/fix_shader_crash

Fix editor crash if passing index as variable to function parameter

akien-mga committed:
Use Ref<T> references as iterators where relevant

And const when possible.

fabriceci committed:
Allow to have multiple collision direction at the same time

YeldhamDev committed:
Make `switch_on_hover` work on embedded windows

akien-mga committed:
Merge pull request #50713 from SaracenOne/drag_and_drop_improvements

Tweak behaviour of drag and drop for 3D scenes to make them less glitchy

Calinou committed:
Merge pull request #38880 from aaronfranke/timer

Use doubles for time everywhere in Timer/SceneTree

akien-mga committed:
Merge pull request #50899 from akien-mga/refref

Use Ref<T> references as iterators where relevant

akien-mga committed:
Merge pull request #50904 from YeldhamDev/switch_hover_embedded

Make `switch_on_hover` work on embedded windows

Chaosus committed:
Fix icon colors in 3d editor on theme changing

raulsntos committed:
Use C++ iterators in the Mono module

Calinou committed:
Merge pull request #50891 from Vitika9/50852

Removed redundant assignment of `blur_pipeline`

raulsntos committed:
Fix Set range iterator implementation

neikeq committed:
Merge pull request #50918 from raulsntos/fix-39432

Ignore paths with invalid chars in PathWhich

timothyqiu committed:
Add check to internal methods to prevent crash

akien-mga committed:
Merge pull request #50925 from timothyqiu/internal-checks

Add checks to internal methods to prevent crash

akien-mga committed:
Merge pull request #50917 from raulsntos/more-iterators

fire committed:
Merge pull request #50355 from Calinou/immediategeometry3d-remove-remains

Remove the remains of ImmediateGeometry3D

vitika9 committed:
Removed Redundant assignement of name inside configure_joypad function

akien-mga committed:
Merge pull request #50928 from Vitika9/50850

Removed Redundant assignement of name inside configure_joypad function

Faless committed:
[Net] Implement RPC channels in MultiplayerAPI.

Faless committed:
[Core] Make enum variant cast and encoding 64 bits

This should fix various issues where retrieving enum values from
scripting languages would result in corrupted values (where 32 bits
were valid, and the other 32 random data).

JFonS committed:
Fixes to editor subgizmos

* Fixed subgizmo editing on scaled nodes.
* Added more clarifications on the coordinate space of subgizmos.
* Given input priority to the transform gizmo over subgizmo selection.

vitika9 committed:
Fixed Camera2D's reset_smoothing() does not work as described

akien-mga committed:
Merge pull request #50912 from Chaosus/fix_editor_3d_icon_theme_changing

Fix icon colors in 3d editor on theme changing

akien-mga committed:
Merge pull request #50933 from JFonS/gizmo_rework_improvements

akien-mga committed:
Merge pull request #50913 from Razoric480/lsp-SymbolKind-fix

Fix LSP reporting wrong types

akien-mga committed:
Merge pull request #50836 from Calinou/3d-editor-camera-preview-add-shortcut

Add shortcut to toggle the 3D editor's camera preview

vnen committed:
Change Variant bool conversion to uint8_t

This ensures more portable conversion since not every path assume bool
is 32-bits and there's no loss converting to 8-bits anyway.

akien-mga committed:
Merge pull request #50931 from Faless/fix/4.x_enum_cast

[Core] Make enum variant cast and encoding 64 bits

akien-mga committed:
Merge pull request #50939 from vnen/fix-variant-bool-conversion

Change Variant bool conversion to uint8_t

BastiaanOlij committed:
Porting cubemap compute shaders to raster for the mobile renderer

m4gr3d committed:
Fix custom build export

akien-mga committed:
Merge pull request #50946 from m4gr3d/fix_custom_build_export_master

Fix custom build export

akien-mga committed:
Merge pull request #50905 from fabriceci/fix-multiple-direction-collision

Allow multiple collision direction at the same time

raulsntos committed:
Use `allowEmpty` parameter in Split

YeldhamDev committed:
Multiple cosmetic fixes for embedded windows

akien-mga committed:
Merge pull request #49723 from aaronfranke/fix-csg-sphere

Fix CSGSphere3D mesh creation

akien-mga committed:
Merge pull request #50677 from Calinou/doc-image-save-exr-editor-only

Document `Image.save_exr()` only being available in editor builds

akien-mga committed:
Merge pull request #50826 from Calinou/editor-improve-mesh-uv-preview

Improve MeshInstance3D UV preview in the editor

akien-mga committed:
Merge pull request #50597 from Calinou/3d-editor-improve-manipulation-gizmo

Improve the 3D editor manipulation gizmo

akien-mga committed:
Merge pull request #50577 from Calinou/2d-editor-ruler-improve-display

Improve the 2D editor ruler display

groud committed:
Implements TileMap layers and move TileSetPlugins's functions to the TileMap node instead

akien-mga committed:
Merge pull request #50961 from YeldhamDev/embed_windows_fixes

Multiple cosmetic fixes for embedded windows

Calinou committed:
Add node configuration warning for GeometryInstance3D distance

Calinou committed:
Use bullet points in the editor instead of dashes where relevant

Calinou committed:
Expose Vulkan's clustered and mobile backends in the project manager

Since OpenGL will not be available in Godot 4.0, this exposes a
choice between Vulkan clustered and Vulkan mobile in the project manager.

Despite the name, Vulkan mobile has many benefits on desktop platforms.
It provides better performance on simple scenes, and ensures that you
won't accidentally use unsupported features while testing your project
on desktop platforms.

The Vulkan backend setting was made into a "basic" setting so that
it can be changed without having to enable the Advanced Settings toggle.

This also improves list formatting to use bullet points and tweaks
the property hint to be more descriptive.

raulsntos committed:
Fix forward calculation in PathFollow3D for the position at the end of the curve

SirQuartz committed:
Fix duplicate shortcut in TileMap Editor

This pull request fixes an issue where the paint and erase tools in the TileMap editor had the same shortcut (E). The erase tool having "E" be its shortcut makes more sense than the paint tool having that be its shortcut. So I changed the paint tool's shortcut to be "D" since nothing else uses it and it's short for "draw", it's also right next to "S" on the keyboard which happens to be the selection tool.

Faless committed:
[Net] Fix RPC ID encoding/decoding for Node methods.

akien-mga committed:
Merge pull request #50991 from Faless/mp/4.x_rpc_config_fix

[Net] Fix RPC ID encoding/decoding for Node methods.

akien-mga committed:
Merge pull request #50981 from Calinou/geometryinstance3d-distance-add-configuration-warnings

Add node configuration warning for GeometryInstance3D distance

timothyqiu committed:
Hide open doc button when not inspecting anything

foxydevloper committed:
Name nodes added from drag & drop by `name_casing`

akien-mga committed:
Merge pull request #50994 from timothyqiu/open-docs-button

Disable open doc button when not inspecting anything

akien-mga committed:
Merge pull request #50989 from SirQuartz/patch-35

Fix duplicate shortcut for paint and erase in TileMap editor

akien-mga committed:
Add script to make source tarball with `.git/HEAD`

This allows to generate the `VERSION_HASH` constant with the Git commit hash
even when building from a source tarball which is not a Git repository (and
without dependency on Git itself).

akien-mga committed:
Allow overriding `VERSION_STATUS` with `GODOT_VERSION_STATUS` in env

`VERSION_STATUS` is part of what constitutes the reference version for a given
Godot build, and is part of the version check for compatible export templates.

For dev snapshots (alpha, beta, RCs), we usually set the `VERSION_STATUS` to
a specific build number (e.g. `beta2`), but this change doesn't end up
committed to the Git repository as we don't want to keep changing `version.py`
for testing builds.

So this new environment override will be what can be used in official builds
and by users making custom builds for specific snapshots.

akien-mga committed:
Merge pull request #50710 from Faless/enet/4.x_refactor

foxydevloper committed:
Make drag and drop into viewport add to root node by default

When dragging and dropping a texture, mesh, or scene from the FileSystem into the
2D or 3D viewport, it will be added as a child of the current scene's root node.

raulsntos committed:
Simplify C# print methods

- Extracts the parameters logic to a single method
- Simplify the handling of null parameters

raulsntos committed:
Use C# interpolated strings

Uses interpolated strings wherever possible.
String concatenations are still left where used for breaking long lines.

akien-mga committed:
Merge pull request #51001 from akien-mga/make_tarball-git-HEAD

Add script to make source tarball with `.git/HEAD`

akien-mga committed:
Merge pull request #51002 from akien-mga/version-status-env-override

Allow overriding `VERSION_STATUS` with `GODOT_VERSION_STATUS` in env

YuriSizov committed:
Fix a crash when trying to load a WebP StreamTexture2D

timothyqiu committed:
Remove colon from column titles

akien-mga committed:
Merge pull request #51015 from pycbouh/core-texture-webp-crash

Fix a crash when trying to load a WebP `StreamTexture2D`

YeldhamDev committed:
Make some small tweaks to the MIME info

Chaosus committed:
Clear debug process id at `stop()` to prevent invalid checking of them

akien-mga committed:
Merge pull request #51018 from timothyqiu/colon-in-column

Remove colon from column titles

akien-mga committed:
Merge pull request #51019 from YeldhamDev/mime_tweaks

Make some small tweaks to the MIME info

akien-mga committed:
Merge pull request #51024 from Chaosus/clear_processes_ids

Clear debug process identifiers array at `stop()` to prevent invalid checking of them

merumelu committed:
makerst: use link titles for external tutorials

jordo committed:
Websocket peer outbound buffer fixes.  Expose outbound buffered amount.

akien-mga committed:
Merge pull request #51022 from raulsntos/csharp-interpolated-strings

Use C# interpolated strings

akien-mga committed:
Merge pull request #51009 from raulsntos/fix-csharp-split

Use `allowEmpty` parameter in Split

YeldhamDev committed:
Properly tag project files as a subclass of plain text in the MIME info

pouleyKetchoupp committed:
Refactor layer property editor grid

- Now able to display up to 32 layers in physics (still 20 for render)
- Adjustable grid size to fit available space in dock
- Expansion icon to display more layers vertically
- Layer numbers in cells to help with selection

foxydevloper committed:
Name new resource files with `snake_case`

nikitalita committed:
Fix binary resource loading and saving

Faless committed:
[Net] Fix ENet 'connect_to_host' creating only one channel.

Passing `0` to `enet_host_create` will allow the maximum amount of
channel supported by ENet. For some reasons, `connect_to_host` will
instead only create 1 channel when passed `0`.
This commit normalize the behaviour to always allocate the maximum
allowed channels when passing `0`.

nikitalita committed:
Use constant for reserved field count

nikitalita committed:
skip uid field length in binary resource if not used

fire committed:
In glTF2 animations, log spam less when running.

fire committed:
glTF2 fallback load PNG and JPG

akien-mga committed:
Merge pull request #51038 from YeldhamDev/mime_yet_again

Properly tag project files as a subclass of plain text in the MIME info

akien-mga committed:
Merge pull request #51011 from foxydevloper/drag-drop-root-default

akien-mga committed:
Merge pull request #51051 from V-Sekai/gltf-less-logs

In glTF2 animations, log spam less when running.

akien-mga committed:
Merge pull request #49149 from YeldhamDev/auto_translate_toggle

Add `auto_translate` toggle for automatic translation

akien-mga committed:
Merge pull request #50982 from Calinou/editor-use-bullet-points

Use bullet points in the editor instead of dashes where relevant

raulsntos committed:
Add Input documentation for `exact_match`

akien-mga committed:
Merge pull request #48567 from Calinou/editor-theme-less-opaque-viewport-focus

Make the focus outline translucent for editor viewports

akien-mga committed:
Merge pull request #35891 from Calinou/editor-viewport-highlight-context-menus

Highlight context menu items at the top of the 2D/3D viewports

akien-mga committed:
Merge pull request #51064 from raulsntos/doc-input-exact_match

Add Input documentation for `exact_match`

akien-mga committed:
doc: Use self-closing tags for `return` and `argument`

For the time being we don't support writing a description for those, preferring
having all details in the method's description.

Using self-closing tags saves half the lines, and prevents contributors from
thinking that they should write the argument or return documentation there.

Faless committed:
[Net] Fix Marshalls infinite recursion crash.

Variants like dictionaries and arrays can have cyclic references, which
caused `encode_variant` to run an infinite recursion.
Instead of keeping a stack and looking for cyclic references which would
make serialization slower, this commit adds a `MAX_RECURSION_DEPTH`
constant to Variant, and have `encode_variant` keep track of the current
recursion depth, bailing when it's too high since this likely means a
cyclic reference has been encountered.

akien-mga committed:
Merge pull request #51041 from foxydevloper/new-resources-snake-case

Name new resource files with `snake_case`

akien-mga committed:
Merge pull request #51066 from Faless/enet/4.x_channels_fix

[Net] Fix ENet 'connect_to_host' creating only one channel.

akien-mga committed:
Merge pull request #51067 from akien-mga/doc-return-argument-self-closing-tags

doc: Use self-closing tags for `return` and `argument`

akien-mga committed:
Merge pull request #51052 from V-Sekai/gltf-fallbacks

glTF2 fallback load PNG and JPG

akien-mga committed:
Merge pull request #51068 from Faless/net/4.x_marshalls_recurse

[Net] Fix Marshalls infinite recursion crash.

YeldhamDev committed:
Fix "Custom *" theme sections being inside "Auto Translate"

akien-mga committed:
Merge pull request #51073 from YeldhamDev/oops

Fix "Custom *" theme sections being inside "Auto Translate"

pouleyKetchoupp committed:
Optimize theme change in code editor

Postpone applying the whole theme when a setting changes, to avoid
updating everything many times when the whole editor theme is changed.

timothyqiu committed:
Do nothing when dragging CSGBox handle perpendicular to the camera

akien-mga committed:
Merge pull request #50984 from Calinou/project-manager-vulkan-clustered-mobile

Expose Vulkan's clustered and mobile backends in the project manager

akien-mga committed:
Merge pull request #51078 from nekomatata/code-editor-theme-change-optimization

Optimize theme change in code editor

SirQuartz committed:
Make "Find in Files" searches ignore directories with `.gdignore` files in them

This pull request fixes an issue where searches using the "Find in Files" function would include folders with `.gdignore` files in them. The editor is supposed to ignore directories with these files in them altogether.

akien-mga committed:
Merge pull request #51080 from timothyqiu/csgbox-nan

Do nothing when dragging CSGBox handle perpendicular to the camera

aaronfranke committed:
Remove obsolete "dectime" method

Replaced by "move_toward"

akien-mga committed:
VariantParser: Fix uninitialized ResourceParser funcs

They could cause a segfault when parsing values with ID "Resource"
as apparently we never set a valid `func` for it.

Fixes crash part of #42115.

akien-mga committed:
Merge pull request #51081 from SirQuartz/patch-38

Make "Find in Files" ignore directories with `.gdignore` in them

akien-mga committed:
Merge pull request #51057 from V-Sekai/bake-reset-anim

Bake RESET animation.

akien-mga committed:
Merge pull request #50935 from Vitika9/50807

akien-mga committed:
Merge pull request #51083 from akien-mga/variantparser-fix-crash-42115

VariantParser: Fix uninitialized ResourceParser funcs

qarmin committed:
Fix 'Attempted to remove invalid ID' errors

RandomShaper committed:
Fix thread start with no user data when target has no default argument

akien-mga committed:
Merge pull request #51084 from aaronfranke/no-dectime

Remove obsolete `dectime` method

akien-mga committed:
Merge pull request #46270 from RandomShaper/fix_can_reset

Rationalize property reversion

Calinou committed:
Print the Vulkan device name in release builds too

This is important information to include in bug reports for exported
projects, and is consistent with the behavior found in the GLES3 and
GLES2 renderers in `3.x`.

Calinou committed:
Use title case instead of uppercase for Vulkan Intel GPU detection

This matches how the vendor name is displayed in most places.

The Apple GPU vendor was also added for the M1.

timothyqiu committed:
Make action names translatable

Calinou committed:
Print a warning when the engine is started as `root`/superuser

This is a security risk (especially when hosting a game server)
and can make audio non-functional on Linux.

Calinou committed:
Move Vulkan debugging prints to verbose

These messages can now be displayed in release builds if the
`--verbose` command line argument is specified, which is useful for
troubleshooting.

akien-mga committed:
Merge pull request #51091 from qarmin/attempted_to_free

Fix 'Attempted to remove invalid ID' errors

YuriSizov committed:
Put multiple colons back into translated strings

Chaosus committed:
Fix crash on doc dragging in script list panel

akien-mga committed:
Merge pull request #51099 from Calinou/vulkan-print-device-release

Print the Vulkan device name in release builds too

akien-mga committed:
Merge pull request #51108 from Calinou/vulkan-print-verbose

Move Vulkan debugging prints to verbose

akien-mga committed:
Merge pull request #51105 from timothyqiu/i18n-action-names

Make action names translatable

YuriSizov committed:
Fix theming in the audio bus editor

akien-mga committed:
Merge pull request #51103 from Calinou/vulkan-detect-intel-title-case

Use title case instead of uppercase for Vulkan Intel GPU detection

akien-mga committed:
Merge pull request #51110 from pycbouh/editor-put-the-colon-back-in

Put multiple colons back into translated strings

akien-mga committed:
Merge pull request #50625 from nekomatata/body-one-direction-layers

One-directional collision layer check for rigid bodies and soft bodies

akien-mga committed:
Merge pull request #51114 from pycbouh/editor-fix-audio-bus-colors

Fix theming in the audio bus editor

SirQuartz committed:
Make the "View" menu in 3D view toolbar stay open when selecting a checkbox

This pull request fixes an issue where the "View" menu in the 3D view toolbar would close when you selected either the "View Origin" or "View Grid" checkboxes. This was inconvenient and wasted time by making you have to reopen the menu in order to get to other settings anytime you changed this.

YuriSizov committed:
Make theme item overrides more obvious in the Inspector

YeldhamDev committed:
Fix being able to add duplicate files in the Localization editor

timothyqiu committed:
Make progress and errors translatable when exporting to Android

akien-mga committed:
Windows: Fix build with SCons 4.2.0

akien-mga committed:
Merge pull request #51124 from YeldhamDev/local_editor_dup_fix

Fix being able to add duplicate files in the Localization editor

akien-mga committed:
Merge pull request #51136 from akien-mga/scons-4.2.0

Windows: Fix build with SCons 4.2.0

akien-mga committed:
Merge pull request #51036 from winterpixelgames/master-ws-fix

WebsocketPeer outbound buffer fixes and buffer size query

akien-mga committed:
Merge pull request #48958 from kleonc/astar-get_available_point_id-start-from-zero

Astar::get_available_point_id Return 0 instead of 1 when empty

Chaosus committed:
Prevent warning spam to console when dragging a CanvasItem in container

akien-mga committed:
Merge pull request #51130 from timothyqiu/i18n-android-export

Make progress and errors translatable when exporting to Android

akien-mga committed:
Merge pull request #51112 from Chaosus/fix_doc_drag_crash

Fix crash on doc dragging in script list panel

akien-mga committed:
Merge pull request #51139 from Chaosus/fix_canvasitem_drag_warning

Prevent warning spam to console when dragging a CanvasItem in container

Chaosus committed:
Makes dictionary instead of string for visual shader version


Update doc/classes/VisualShader.xml

Co-authored-by: Hugo Locurcio <hugo.locurcio@hugo.pro>

akien-mga committed:
Merge pull request #50942 from BastiaanOlij/cubemap_raster

Porting cubemap compute shaders to raster for the mobile renderer

akien-mga committed:
Merge pull request #51119 from pycbouh/control-theme-overrides-rename

Make theme item overrides more obvious in the Inspector

akien-mga committed:
Dependency editor: Edit message about move to trash

Fixes #47106.

akien-mga committed:
Merge pull request #50430 from goostengine/ctrl-enter-commit

VCS: Allow to stage and commit all files with Ctrl + Enter

timothyqiu committed:
Use Command+Drag in EditSpinSlider on macOS

akien-mga committed:
Merge pull request #49783 from aaronfranke/real-editor-plugins

Use real_t in editor plugins

Faless committed:
[Marshalls] Fix Float64Array and Int64Array serialization.

One was incorrectly reading the size (potentially causing out-of-buffer
read), the other also potentially causing out-of-buffer write during
encoding.

RandomShaper committed:
Fix up property reversion

KoBeWi committed:
Add null check to Tween.bind_node()

kleonc committed:
TabContainer Fix moving dropped tab at incorrect child index

akien-mga committed:
Merge pull request #51170 from timothyqiu/spin-slider-osx

Use Command+Drag in EditSpinSlider on macOS

akien-mga committed:
Merge pull request #51172 from Faless/io/4.x_64_array_marshalls

[Marshalls] Fix Float64Array and Int64Array serialization.

akien-mga committed:
Merge pull request #51176 from KoBeWi/nullbind

Add null check to Tween.bind_node()

akien-mga committed:
Merge pull request #51173 from RandomShaper/tidy_property_reversion

Fix up property reversion

timothyqiu committed:
Validate TileData enum range before using it as index

timothyqiu committed:
Fix crash when number of indices to optimize is not a multiple of 3

akien-mga committed:
Merge pull request #51004 from groud/tilemap_layers

Implements TileMap layers and move TileSetPlugins's functions to the TileMap node instead

akien-mga committed:
Merge pull request #51185 from timothyqiu/tile-data-enum-range

Validate TileData enum range before using it as index

timothyqiu committed:
Validates VisualScript.add_node input node

timothyqiu committed:
Validates set_sdfgi_cascades argument range

akien-mga committed:
Merge pull request #51190 from timothyqiu/vs-node

Validates VisualScript.add_node input node

akien-mga committed:
Merge pull request #51187 from timothyqiu/vertex-index-3

Fix crash when number of indices to optimize is not a multiple of 3

akien-mga committed:
Merge pull request #51192 from timothyqiu/sdfgi-enum-range

Validates set_sdfgi_cascades argument range

Faless committed:
[Net] Fix IP address resolution incorrectly locking the main thread.

This seems to be a pretty old bug, older then originally reported (at
least under certain circumstances).

The IP singleton uses a resolve queue so developers can queue hostnames
for resolution in a separate while keeping the main thread unlocked
(address-resolution OS functions are blocking, and could block for a long
time in case of network disruption).

In most places though, the address resolution function was called with
the mutex locked, causing other functions (querying status, queueing
another hostname, ecc) to block until that resolution ended.

This commit ensures that all calls to OS address resolution are done
with the mutex unlocked.

akien-mga committed:
Merge pull request #50122 from Paulb23/code_edit_auto_brace_completion

akien-mga committed:
Merge pull request #51168 from akien-mga/dependency-move-to-trash

Dependency editor: Edit message about move to trash

Chaosus committed:
Merge pull request #51144 from Chaosus/vs_version

Makes dictionary instead of string for visual shader version

akien-mga committed:
Merge pull request #51039 from nekomatata/layer-grid-32

Refactor layer property editor grid

Calinou committed:
Improve CodeEdit's callhint highlight visibility

This makes the currently edited parameter more visible in the
script editor's code completion hint.

fire committed:
Merge pull request #51042 from nikitalita/fix_binary_res_load_save

Fix binary resource loading and saving

Commit:
Grammar fix.

akien-mga committed:
Merge pull request #51198 from Calinou/codeedit-improve-callhint-highlight

Improve CodeEdit's callhint highlight visibility

akien-mga committed:
Merge pull request #51203 from kbsanders/patch-1

Grammar fix.

akien-mga committed:
Merge pull request #50703 from Chaosus/vs_remove_subparticles_from_start

Removes EmitParticle node from the start functions in visual shader

akien-mga committed:
Merge pull request #51093 from RandomShaper/fix_thread_null_ud

 Fix thread start with no user data when target has no default argument

Chaosus committed:
Merge pull request #50966 from Chaosus/vs_builtins

akien-mga committed:
Merge pull request #50986 from raulsntos/fix-path-3d-end

akien-mga committed:
Merge pull request #51177 from kleonc/tab_container-drop-tab-index-fix

TabContainer Fix moving dropped tab to incorrect child index

akien-mga committed:
Merge pull request #51118 from SirQuartz/patch-39

Make the "View" menu in the 3D viewport stay open when selecting a checkbox

akien-mga committed:
Merge pull request #51107 from Calinou/warn-engine-run-as-root

Print a warning when the engine is started as `root`/superuser

akien-mga committed:
Merge pull request #51006 from foxydevloper/drag-drop-naming

Name nodes added when drag & dropping an image by `name_casing`

akien-mga committed:
Merge pull request #51033 from merumelu/makerst-tutorial-links

makerst: use link titles for external tutorials

akien-mga committed:
i18n: Sync translations with Weblate

Last POT sync with the `3.3` branch, we'll now switch Weblate to track
`3.x` to prepare for the 3.4 release.

(cherry picked from commit dec840452d5986ec8099b92ebabf454757da8b04)

akien-mga committed:
i18n: Sync template with current 3.x codebase

Weblate will now track the state of `3.x` to prepare for the 3.4 release.

(cherry picked from commit 92bafdcafb2c2d50b9079728842b825c72b99c19)

akien-mga committed:
Merge pull request #51199 from Faless/net/4.x_ip_lock

akien-mga committed:
Merge pull request #50893 from KoBeWi/how_to_config_file

Improve ConfigFile example

YuriSizov committed:
Cache EditorResourcePicker's allowed types

akien-mga committed:
Merge pull request #51213 from pycbouh/editor-resource-picker-cache

Cache `EditorResourcePicker`'s allowed types

Faless committed:
[Headless] Add --headless switch (no rendering, no audio).

Also remove now unused "--no-window" option, and relative OS getter and
setter.

Faless committed:
Merge pull request #50454 from Ev1lbl0w/gsoc21-dap

Implemented initial DAP support

Faless committed:
Merge pull request #51216 from Faless/headless/switch

[Headless] Add --headless switch (no rendering, no audio).

YuriSizov committed:
Fix the editor theme application for the Mono build log

akien-mga committed:
Merge pull request #51220 from pycbouh/editor-mono-build-log

Fix the editor theme application for the Mono build log

neikeq committed:
Merge pull request #51020 from raulsntos/csharp-print-methods

Simplify C# print methods

pouleyKetchoupp committed:
Fix inverted condition in CharacterBody2D/3D set_max_slides

timothyqiu committed:
Disable save button when Inspector is not editing anything

neikeq committed:
Merge pull request #47691 from 31/fix-script_class-null-master

Fix `script_class` null access when reloading a deleted C# script

akien-mga committed:
Merge pull request #51226 from nekomatata/fix-max-slides-error

Fix inverted condition in CharacterBody2D/3D set_max_slides

akien-mga committed:
Merge pull request #51228 from timothyqiu/inspector-update

Disable save button when Inspector is not editing anything

akien-mga committed:
Merge pull request #51005 from Faless/mp/4.x_channels

[Net] Implement RPC channels in MultiplayerAPI.

YuriSizov committed:
Improve user communications in the theme editor

akien-mga committed:
Merge pull request #51242 from pycbouh/editor-theme-instructions

Improve user communications in the theme editor

YuriSizov committed:
Add theme item descriptions to the online documentation

akien-mga committed:
Merge pull request #48615 from Razoric480/lsp-rename

Implement LSP didSave notification and rename request

akien-mga committed:
Merge pull request #51247 from pycbouh/docs-extract-theme-items

Add theme item descriptions to the online documentation

mechPenSketch committed:
Merge branch 'master' of https://github.com/mechPenSketch/godot
